vercel/resource_deployment.go

```diff
@@ -308,7 +308,26 @@ func (r resourceDeployment) Update(ctx context.Context, req tfsdk.UpdateResource
 // Delete does nothing other than clear the existing terraform state for a Deployment.
 // This is done intentionally, as typically, Vercel users do not continually delete old Deployments.
 func (r resourceDeployment) Delete(ctx context.Context, req tfsdk.DeleteResourceRequest, resp *tfsdk.DeleteResourceResponse) {
-	tflog.Trace(ctx, "deleted deployment")
+	//err := r.p.client.DeleteProject(ctx, state.ID.Value, state.TeamID.Value)
```

Comment:
I'm hesitant to merge this just because the comment here (L309) kind of conflicts with the implementation. I'm really unsure whether people will want deployments deleted. 

AFAIK it's a fairly rare operation to perform on Vercel, with the majority of people leaving deployments around for later reference. It may be better to include some optional flag on the terraform state to indicate that the deployments should be deleted though, allowing this to be an opt-in behaviour.

Reply:
What about making it optional? based on an input parameter for example? We use vercel deployments to make very ephemeral e2e tests (along with other purposes) so we create a lot of ephemeral deployments. As a Devops engineer I'm kinda inclined to delete them.
I'll make this optional.

Reply:
And the default will be to leave them deployed.

Reply:
That would be 👍

Reply:
I changed the behaviour to be optional and default to false